packages/core/src/__tests__/data/angular/dynamic-component.raw.tsx

```diff
@@ -0,0 +1,24 @@
+import { useState } from '@builder.io/mitosis';
+
+export default function MyComponent(props) {
+  const [obj, setObj] = useState({
+    name: 'foo',
+    Component: FooComponent,
+  });
+
+  function onClick() {
+    console.log('hello');
+  }
+
+  return (
+    <obj.Component
+      hello="world"
+      onClick={() => onClick()}
```

Comment:
can you add another example that receives an argument? I want to see if that still works with your code

Something like:

```suggestion
      onClick={() => onClick()}
      onChange={(event) => onClick(event)}
```

Reply:
sure!

this will not work I think as they are just passed as function and we cannot inline functions directly in angular template:

for example here we had to manually call the function with event: 

https://github.com/BuilderIO/builder/pull/3237/files#diff-12898be32207d680f53e79a6999052f6017b1b9f9069019aa778c9fdee649f07R17

Reply:
ok. that should get resolved once we cleanup the angular template workarounds as I described to you

Reply:
yes!!